code/datums/traits.dm

```diff
@@ -0,0 +1,79 @@
+/datum/proc/add_trait(trait, source)
+	if(isnull(status_traits))
+		status_traits = list()
+		
+	if(!status_traits[trait])
+		status_traits[trait] = list(source)
+	else
+		status_traits[trait] |= list(source)
+		
+
+/datum/proc/remove_trait(trait, list/sources, force)
+	if(isnull(status_traits))
```

Comment:
Why create the list if it's empty, won't that just create more work for later calls of the same proc? Ditto the other three methods

Reply:
oh yeah, you're right